mmv1/third_party/terraform/services/composer/resource_composer_environment.go.tmpl

```diff
@@ -725,7 +725,7 @@ func ResourceComposerEnvironment() *schema.Resource {
 													Type:         schema.TypeString,
 													Optional:     true,
 													ValidateFunc: validation.StringInSlice([]string{"CLOUD_LOGGING_ONLY", "CLOUD_LOGGING_AND_CLOUD_STORAGE"}, false),
-													Description:  `Whether logs in cloud logging only is enabled or not. This field is supported for Cloud Composer environments in versions composer-2.0.32-airflow-2.1.4 and newer.`,
+													Description:  `Whether logs in cloud logging only is enabled or not. This field is supported for Cloud Composer environments in versions composer-2.0.32-airflow-2.1.4 and newer but not in composer-3*`,
```

Comment:
You'll also need to update the handwritten doc file mmv1/third_party/terraform/website/docs/r/composer_environment.html.markdown, given this is a handwritten resource and the documentation is not auto-generated.

Reply:
I have replaced `task_logs_retention_config` with `airflow_metadata_retention_config` for Composer 3 section.

Reply:
Thanks! Just want to confirm if I understand correctly, `task_logs_retention_config` is supported for Composer 2 only while `airflow_metadata_retention_config` is for Composer 3 only

Reply:
Yes, I have tried to create a Composer 3 cluster with it and it fails to do so with 400 error. It is [documented that Composer 3 do not support storing logs in bucket](https://cloud.google.com/composer/docs/composer-3/enable-saving-logs-to-environment-bucket).